src/lib/cfa/sudoers.rb

```diff
@@ -0,0 +1,237 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/target_file"
+require "cfa/base_model"
+require "cfa/augeas_parser"
+require "cfa/matcher"
+require "cfa/placer"
+
+module CFA
+  # Model to handle users specifications in the /etc/sudoers file
+  #
+  # @example Loading specifications
+  #   file = Sudoers.new
+  #   file.load
+  #   file.current_user_specs
+  #
+  # @example Shortcut loading specificitions
+  #   Sudoers.load.users_specs
+  #
+  # @example Writing new specifications
+  #   file = Sudores.new
+  #   file.load
+  #   file.clean_users
+  #   file.add_user({ "user" => "yast", "host" => "ALL", "commands" => ["/sbin/yast2"] })
+  #   file.add_user({ "user" => "foo", "host" => "ALL", "commands" => ["ALL"], "tag" => "NOPASSWD" })
+  #   file.save
+  class Sudoers < BaseModel
+    # @return [AgueasParser] the Augeas parser for sudoers file
+    PARSER = AugeasParser.new("sudoers.lns")
+    private_constant :PARSER
+
+    # @return [String] default path to the sudoers file
+    DEFAULT_PATH = "/etc/sudoers".freeze
+    private_constant :DEFAULT_PATH
+
+    # @return [Matcher] matcher for the users collection
+    USERS_MATCHER = Matcher.new(collection: "spec")
+    private_constant :USERS_MATCHER
+
+    class << self
+      # Instantiates and loads a file
+      #
+      # This method is basically a shortcut to instantiate and load the content in just one call.
+      #
+      # @param file_handler [#read,#write] something able to read/write a string (like File)
+      # @return [Sudoers] File with the already loaded content
+      def load(file_path: DEFAULT_PATH, file_handler: Yast::TargetFile)
+        new(file_path: file_path, file_handler: file_handler).tap(&:load)
+      end
+    end
+
+    # Constructor
+    #
+    # @param file_handler [#read,#write] something able to read/write a string (like File)
+    #
+    # @see CFA::BaseModel#initialize
+    def initialize(file_path: DEFAULT_PATH, file_handler: nil)
+      super(PARSER, file_path, file_handler: file_handler)
+    end
+
+    # Read and prepare the data
+    #
+    # @see CFA::BaseModel#initialize
+    # @see #fix_collection_names
+    # @see #set_start_placer
+    def load
+      super
+
+      fix_collection_names(data)
+      @start_placer = set_start_placer
+    end
+
+    # Return all users specifications
+    #
+    # @return [Array<Hash>]
+    def users_specs
+      current_users.reduce([]) do |specs, user|
+        spec = {}
+        values = user[:value]
+
+        spec["user"] = values["user"]
+        spec["host"] = values["host_group"]["host"]
```

Comment:
I think we need really good test cases with file content. As I am not sure here if it does not crash if host_group is not specified at all.

Reply:
Removing the _host_ part in the sudoers file result in a `CFA::AugeasParsingError` when trying to `#load`

Reply:
important question is if it is valid sudo configuration ( so bug in augeas lense ) or it really cannot be empty ( then please add comment here ).

Reply:
I need to check it